src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -666,32 +668,33 @@ void Transformations::MainSnippets(void) {
 #endif // OPENVINO_ARCH_X86_64
 
     if (snippetsMode != Config::SnippetsMode::IgnoreCallback) {
-        CPU_SET_CALLBACK_X64(snippetsManager,
-            [&](const std::shared_ptr<const ov::Node>& n) -> bool {
-                // Tranformation callback is called on MatMul0
-                if (!is_supported_matmul(n))
-                    return true;
-                // Search for MatMul1
-                auto child = n->get_output_target_inputs(0).begin()->get_node()->shared_from_this();
-                while (!ov::is_type<const ov::op::v0::MatMul>(child)) {
-                    child = child->get_output_target_inputs(0).begin()->get_node()->shared_from_this();
-                }
-                if (!is_supported_matmul(child))
-                    return true;
-                const auto pshape = child->get_input_partial_shape(0);
-                const auto shape = pshape.get_shape();
-                const auto parallel_work_amount =
-                        std::accumulate(shape.rbegin() + 2, shape.rend(), 1, std::multiplies<size_t>());
-                // Heuristic values:
-                //    parallelism work amount - not enough work amount for parallelism
-                // TODO: The heuristic will be removed after parallelism support on JIT level
-                const auto needed_num_of_threads = 12lu;
-                const auto is_unsupported_parallel_work_amount =
-                    parallel_get_num_threads() / 2 > parallel_work_amount &&
-                    static_cast<size_t>(parallel_work_amount) < needed_num_of_threads;
-                return is_unsupported_parallel_work_amount;
-            },
-            snippets::pass::TokenizeMHASnippets);
+        auto mha_callback = [&](const std::shared_ptr<const ov::Node>& n) -> bool {
+            // Tranformation callback is called on MatMul0
+            if (!is_supported_matmul(n))
```

Comment:
We have `is_supported_matmul()` in TokenizeMHASnippets callback, and then `is_matmul_supported()` inside the transformation. Do you think we can unite these checks?

Reply:
`ov::snippets::pass::TokenizeMHASnippets::is_matmul_supported` contains checks only for the first matmul in MHA pattern whereas `is_supported_matmul()` lambda contains requirements for both matmuls. So I think we don't need to unite these checks.
However, it's good idea to rename static `TokenizeMHASnippets` method to `is_matmul0_supported`: it is done

BTW after offline discussion with @a-sidorova CPU callback's part was refactored: `snippets::pass::ExtractReshapesFromMHA` callback now don't check child matmul because it's not guarantied that we find it in while loop (because the second matmul is not defined in the transformation matcher)